### ISSUE TITLE
get-key-type test fails

### ISSUE THREAD

**Issue by Pilifer** [artyom-poptsov/guile-ssh#12]
While installing AUR package (on Arch with `libssh-0.9.0-1` installed) `get-key-type` test fails.

Here is `test-suite.log`:

```
============================================
   Guile-SSH 0.11.3: tests/test-suite.log
============================================

# TOTAL: 11
# PASS:  10
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: key
=========

Backtrace:
           3 (primitive-load "/home/fjl/.cache/yay/guile-ssh/src/gui…")
In ice-9/eval.scm:
    619:8  2 (_ #(#(#(#<directory (guile-user) 56075d6cb140> #) #) #))
   293:34  1 (_ #(#(#(#<directory (guile-user) 56075d6cb140> #) #) #))
In unknown file:
           0 (public-key->string #<key ecdsa (public) 56075d859da0>)

ERROR: In procedure public-key->string:
Throw to key `guile-ssh-error' with args `("public-key->string" "Unable to convert the key to a string" #<key ecdsa (public) 56075d859da0> #f)'.
%%%% Starting test key  (Writing full log to "key.log")
tests/key.scm:90: FAIL get-key-type
```
and the test:
```
Test begin:
  test-name: "get-key-type"
  source-file: "tests/key.scm"
  source-line: 90
  source-form: (test-assert "get-key-type" (begin (set-log-userdata! "get-key-type") (and (eq? (quote rsa) (get-key-type *rsa-key*)) (eq? (quote ds
s) (get-key-type *dsa-key*)) (when-openssl (eq? (quote ecdsa) (get-key-type *ecdsa-key*))))))
Test end:
  result-kind: fail
  actual-value: #f
```

same content in `key.log`

**Comment by dongcarl** [comment]
Same error here, here's the `key.log`:
https://pastebin.com/E65ehgSv

**Comment by vagrantc** [comment]
these tests also fail on Debian unstable and Debian testing using libssh 0.9.1-1, works with libssh 0.8.7-1.

**Comment by vagrantc** [comment]
Correction, it works with guile-ssh 0.11.3 and libssh 0.8.7 on Debian when also adding two patches from upstream: 

dac7ae6f3347056f46d664f6104cf6425ae25394 
tests/server.scm ("server-set!, valid values"): Add fix for libssh 0.8           

01cafef0dd87b6afda41942070e73b55b45a5ed2 
libguile-ssh: Fix building with libssh 0.8

**Comment by artyom-poptsov** [comment]
Hello!  Currently I'm working on a new and (hopefully) better implementation of tests for Guile-SSH (see  `wip-new-tests` branch).  As soon as the updated test framework start working properly, I'll make a new Guile-SSH release with fixes.

Meanwhile, you should use patches from the upstream for now.

Thanks!

-- Artyom

**Comment by artyom-poptsov** [comment]
Hello!

Could you please check Guile-SSH from the `master` branch?  I added some fixes for keys in 
104d797.

Thanks!

-- Artyom

**Comment by lantw44** [comment]
```
libguile-ssh/Makefile.am:48: error: HAVE_LIBSSH_0_8 does not appear in AM_CONDITIONAL
```
After manually fixing the error, the test passes.

**Comment by artyom-poptsov** [comment]
Thanks, @lantw44!  I fixed that in a15543f.  Does it work for you w/o modifying the code?

-- Artyom

**Comment by lantw44** [comment]
Applying both 104d797096e966e91f777959d7cc5d8831c1a942 and a15543f7114b3f1ade8cce2cd1065e6aa60c14cc fixes the problem, but there is a problem on libssh < 0.9:
```
make[2]: Entering directory '/builddir/build/BUILD/guile-ssh-0.11.3/libguile-ssh'
  CC       libguile_ssh_la-key-type.lo
make[2]: Leaving directory '/builddir/build/BUILD/guile-ssh-0.11.3/libguile-ssh'
key-type.c:48:26: error: 'SSH_KEYTYPE_ECDSA_P256' undeclared here (not in a function); did you mean 'SSH_KEYTYPE_ECDSA'?
   { "ecdsa-p256",        SSH_KEYTYPE_ECDSA_P256        },
                          ^~~~~~~~~~~~~~~~~~~~~~
                          SSH_KEYTYPE_ECDSA
key-type.c:49:26: error: 'SSH_KEYTYPE_ECDSA_P384' undeclared here (not in a function); did you mean 'SSH_KEYTYPE_ECDSA'?
   { "ecdsa-p384",        SSH_KEYTYPE_ECDSA_P384        },
                          ^~~~~~~~~~~~~~~~~~~~~~
                          SSH_KEYTYPE_ECDSA
key-type.c:50:26: error: 'SSH_KEYTYPE_ECDSA_P521' undeclared here (not in a function); did you mean 'SSH_KEYTYPE_ECDSA'?
   { "ecdsa-p521",        SSH_KEYTYPE_ECDSA_P521        },
                          ^~~~~~~~~~~~~~~~~~~~~~
                          SSH_KEYTYPE_ECDSA
key-type.c:51:26: error: 'SSH_KEYTYPE_ECDSA_P256_CERT01' undeclared here (not in a function); did you mean 'SSH_KEYTYPE_RSA_CERT01'?
   { "ecdsa-p256-cert01", SSH_KEYTYPE_ECDSA_P256_CERT01 },
                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                          SSH_KEYTYPE_RSA_CERT01
key-type.c:52:26: error: 'SSH_KEYTYPE_ECDSA_P384_CERT01' undeclared here (not in a function); did you mean 'SSH_KEYTYPE_RSA_CERT01'?
   { "ecdsa-p384-cert01", SSH_KEYTYPE_ECDSA_P384_CERT01 },
                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                          SSH_KEYTYPE_RSA_CERT01
key-type.c:53:26: error: 'SSH_KEYTYPE_ECDSA_P521_CERT01' undeclared here (not in a function); did you mean 'SSH_KEYTYPE_RSA_CERT01'?
   { "ecdsa-p521-cert01", SSH_KEYTYPE_ECDSA_P521_CERT01 },
                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                          SSH_KEYTYPE_RSA_CERT01
make[2]: *** [Makefile:638: libguile_ssh_la-key-type.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
```
https://copr-be.cloud.fedoraproject.org/results/lantw44/guix/fedora-29-x86_64/01034705-guile-ssh/builder-live.log.gz

I think you meant `#if HAVE_LIBSSH_0_9`, not `#ifdef HAVE_LIBSSH_0_9`, right?

**Comment by artyom-poptsov** [comment]
Sure thing, I meant to use `#if` directive here.  Thanks for pointing this out!  Should be fixed in 75cec44.  Could you please check?

-- Artyom

**Comment by lantw44** [comment]
Yes, I confirm that it is fixed now.
